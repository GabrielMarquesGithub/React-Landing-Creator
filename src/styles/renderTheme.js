import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (element) => {
  return render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
};
