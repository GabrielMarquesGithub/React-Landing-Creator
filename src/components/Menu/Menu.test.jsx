import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Menu } from '.';

import linksMock from '../NavLinks/mock';
import { theme } from '../../styles/theme';
const logoData = {
  text: 'text',
  link: '#target1',
  imageUrl: 'assets/images/logo.svg',
};
import 'jest-styled-components';

describe('<Menu />', () => {
  it('should render a logo and a main menu nav', () => {
    const { container } = renderTheme(
      <Menu logoData={logoData} links={linksMock} />,
    );
    expect(screen.getByRole('heading', { name: 'text' })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { name: 'Main menu' }),
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
  it('should render menu mobile and button for open and close the menu', () => {
    renderTheme(<Menu logoData={logoData} links={linksMock} />);
    const button = screen.getByLabelText('Open/Close menu');

    expect(button).toHaveStyle({ display: 'none' });
    expect(button).toHaveStyleRule('display', 'flex', {
      media: `(${theme.media.lessMedium})`,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
    expect(button.nextSibling).toHaveStyleRule('opacity', '0', {
      media: `(${theme.media.lessMedium})`,
    });

    fireEvent.click(button);
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();
    expect(button.nextSibling).toHaveStyleRule('opacity', '1', {
      media: `(${theme.media.lessMedium})`,
    });

    fireEvent.click(button.nextSibling);
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
    expect(button.nextSibling).toHaveStyleRule('opacity', '0', {
      media: `(${theme.media.lessMedium})`,
    });
  });
});
