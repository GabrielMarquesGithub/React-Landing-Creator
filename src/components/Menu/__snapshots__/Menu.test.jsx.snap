// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render a logo and a main menu nav 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  text-transform: uppercase;
  color: #0A1128;
  font-size: 2.4rem;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 img {
  height: 2.5rem;
}

.c10 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #DC143C;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  border-bottom: #DDDDDD;
  background: #ffffff;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.c2 .c3 {
  padding: 0 3.2rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  color: #ffffff;
  background: #0A1128;
  border: none;
  display: none;
}

.c0 > svg {
  width: 3rem;
  height: 3rem;
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c2 {
    visibility: hidden;
    opacity: 0;
    bottom: 0;
  }

  .c2 > .c3 {
    max-height: 100vh;
    overflow-y: auto;
  }
}

@media (max-width:768px) {
  .c5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    gap: 3.2rem;
    padding: 3.2rem;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <section
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
        >
          <a
            class="c7"
            href="#target1"
          >
            <img
              alt="text"
              src="assets/images/logo.svg"
            />
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c8"
        >
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target1"
              target="_self"
            >
              Link 1
            </a>
          </div>
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target2"
              target="_self"
            >
              Link 2
            </a>
          </div>
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target3"
              target="_self"
            >
              Link 3
            </a>
          </div>
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target4"
              target="_self"
            >
              Link 4
            </a>
          </div>
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target5"
              target="_self"
            >
              Link 5
            </a>
          </div>
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target6"
              target="_self"
            >
              Link 6
            </a>
          </div>
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target7"
              target="_self"
            >
              Link 7
            </a>
          </div>
          <div
            class="c9"
          >
            <a
              class="c10"
              href="#target8"
              target="_self"
            >
              Link 8
            </a>
          </div>
        </nav>
      </div>
    </section>
  </div>
</div>
`;
